### ISSUE TITLE
current SSIM regression layer is not written properly

### ISSUE THREAD

**Issue by johnnychen94** [johnnychen94/DnCNN#2]
Current SSIM forward and backward function is written according to
  Zhao, H., Gallo, O., Frosio, I., & Kautz, J. (2017). Loss Functions for Image Restoration With Neural Networks, 3(1), 47–57.
However, this is not the true form of SSIM loss, this should be rewritten. Besides, the network doesn't learn anything during training (probably my fault) using the current version of ssim regression layer code
